File: frontend_app/lib/repository/messages.ts
// メッセージを取得、送信する際の要求関数
import { Message } from "@/types/chat-app"
import { apiGet, apiPost } from "@/lib/backend";

// メッセージリストの取得（api/で定義しているプロキシに送信）
export function listMessages(threadId: string) {
  return apiGet<Message[]>("/api/main/messages", { threadId }); // プロキシに送信
}

// 新規メッセージの追加（api/で定義しているプロキシに送信）
export function createMessage(input: { threadId: string; role: "user" | "assistant"; content: string }) {
  const body = {
    threadId: input.threadId,
    role: input.role,
    content: input.content,
  };
  return apiPost<Message[] | Message>("/api/main/messages", body);  // プロキシに送信
}

File: backend_app/app/routers/messages.py
from fastapi import APIRouter, Depends, Query
from pydantic import BaseModel
from typing import Literal
from crud import SupaRest
from deps import bearer_token

# このファイルに指定されているアドレスは”root/api/v1/messages”が先頭につく
router = APIRouter(prefix="/messages", tags=["messages"])

# ==================================================
## 型定義
# ==================================================
Role = Literal["user", "assistant"]


class MessageCreate(BaseModel):
    threadId: str
    role: Role
    content: str


# ==================================================
## CRUD 定義
# ==================================================
@router.get("")
async def list_messages(threadId: str = Query(...), token: str = Depends(bearer_token)):
    # トークンを入れて認証ユーザとしての通信で使用する関数を群を呼び出す
    client = SupaRest(token)
    params = {
        "select": "*",
        "thread_id": f"eq.{threadId}",
        "order": "created_at.asc",
    }  # クエリの作成
    return await client.get("messages", params=params)  # GETメソッドの発行


@router.post("")
async def create_message(payload: MessageCreate, token: str = Depends(bearer_token)):
    # トークンを入れて認証ユーザとしての通信で使用する関数を群を呼び出す
    client = SupaRest(token)
    body = {
        "thread_id": payload.threadId,
        "role": payload.role,
        "content": payload.content,
    }  # リクエストボディを作成
    return await client.post(
        "messages",
        json=body,
        prefer="return=representation",
        content_profile="app",
    )  # POSTメソッドの発行
